libbeat/processors/xmldecode/xmldecode_test.go

```diff
@@ -0,0 +1,255 @@
+								"title":  "The Recognitions",
+							},
+							map[string]interface{}{
+								"author": "Ralls, Kim",
+								"review": "Some review.",
+								"title":  "Midnight Rain",
+							},
+						},
+					},
+				},
+			},
+			error:        false,
+			errorMessage: "",
+		},
+		{
+			description: "Decoding with multiple xml objects",
+			config: xmlDecodeConfig{
+				Fields: []string{"message"},
+			},
+			Input: common.MapStr{
+				"message": `<?xml version="1.0"?>
+				<catalog>
+					<book>
+					<author>William H. Gaddis</author>
+					<title>The Recognitions</title>
+					<review>One of the great seminal American novels of the 20th century.</review>
+				</book>
+				<book>
+					<author>Ralls, Kim</author>
+					<title>Midnight Rain</title>
+					<review>Some review.</review>
+				</book>
+				<secondcategory>
+					<paper id="bk102">
+						<test2>Ralls, Kim</test2>
+						<description>A former architect battles corporate zombies, 
+						an evil sorceress, and her own childhood to become queen 
+						of the world.</description>
+					</paper>
+				</secondcategory>
+				</catalog>`,
+			},
+			Output: common.MapStr{
+				"message": map[string]interface{}{
+					"catalog": map[string]interface{}{
+						"book": []interface{}{
+							map[string]interface{}{
+								"author": "William H. Gaddis",
+								"review": "One of the great seminal American novels of the 20th century.",
+								"title":  "The Recognitions",
+							},
+							map[string]interface{}{
+								"author": "Ralls, Kim",
+								"review": "Some review.",
+								"title":  "Midnight Rain",
+							},
+						},
+						"secondcategory": map[string]interface{}{
+							"paper": map[string]interface{}{
+								"description": "A former architect battles corporate zombies, \n\t\t\t\t\t\tan evil sorceress, and her own childhood to become queen \n\t\t\t\t\t\tof the world.",
+								"id":          "bk102",
+								"test2":       "Ralls, Kim",
+							},
+						},
+					},
+				},
+			},
+			error:        false,
+			errorMessage: "",
+		},
+		{
+			description: "Decoding with broken XML format, with AddErrorKey enabled",
+			config: xmlDecodeConfig{
+				Fields:      []string{"message"},
+				AddErrorKey: true,
+			},
+			Input: common.MapStr{
+				"message": `<?xml version="1.0"?>
+				<catalog>
+					<book>
+					<author>William H. Gaddis</author>
+					<title>The Recognitions</title>
+					<review>One of the great seminal American novels of the 20th century.</review>
+				</ook>
+				catalog>`,
+			},
+			Output: common.MapStr{
+				"message": (map[string]interface{})(nil),
+				"error":   []string{"error trying to decode XML field xml.Decoder.Token() - XML syntax error on line 7: element <book> closed by </ook>"},
+			},
+			error:        true,
+			errorMessage: "error trying to decode XML field xml.Decoder.Token() - XML syntax error on line 7: element <book> closed by </ook>",
+		},
```

Comment:
Add tests for some of the other options? And a case where the field doesn't exist?